barretenberg/cpp/src/barretenberg/stdlib/primitives/group/cycle_group.cpp

```diff
@@ -447,7 +447,15 @@ template <typename Builder> cycle_group<Builder> cycle_group<Builder>::operator+
     // if x_coordinates match, lambda triggers a divide by zero error.
     // Adding in `x_coordinates_match` ensures that lambda will always be well-formed
     auto x_diff = x2.add_two(-x1, x_coordinates_match);
-    auto lambda = (y2 - y1) / x_diff;
+    // Computes lambda = (y2-y1)/x_diff, using the fact that x_diff is never 0
+    field_t lambda;
+    if ((!y2.is_constant() || !y1.is_constant()) && !x_diff.is_constant()) {
+        lambda = field_t::from_witness(context, (y2.get_value() - y1.get_value()) / x_diff.get_value());
+        field_t::evaluate_polynomial_identity(x_diff, lambda, -y2, y1);
+    } else {
+        lambda = (y2 - y1).divide_no_zero_check(x_diff);
+    }
```

Comment:
```suggestion
    if (x_diff.is_constant() || (y2.is_constant() && y1.is_constant())) {
        lambda = (y2 - y1).divide_no_zero_check(x_diff);
    } else {
         lambda = field_t::from_witness(context, (y2.get_value() - y1.get_value()) / x_diff.get_value());
         field_t::evaluate_polynomial_identity(x_diff, lambda, -y2, y1);
    }
```

This if condition is pretty complex and can be simplified by swapping the two branches.

Reply:
Done